packages/codegen-ui-react/lib/react-studio-template-renderer.ts

```diff
@@ -1296,5 +1302,31 @@ export abstract class ReactStudioTemplateRenderer extends StudioTemplateRenderer
     return undefined;
   }
 
+  /* To map predefined props to children props for variants
+   *
+   * Example: Text prop label is mapped to to Text prop Children
+   *
+   */
+  private mapSyntheticPropsForVariants() {
+    if (!isStudioComponentWithVariants(this.component)) {
+      return;
+    }
+
+    this.component.variants.forEach((variant) => {
+      // loop through the keys in the dict. Ex. button, button.text
+      Object.keys(variant.overrides).forEach((overridesKey) => {
+        const childrenPropMapping = PrimitiveChildrenPropMapping[Primitive[overridesKey as Primitive]];
+        if (childrenPropMapping !== undefined) {
+          const propsInOverrides = variant.overrides[overridesKey];
+          // only remap if children prop is not defined in this particular overrides section
+          if (propsInOverrides.children === undefined && propsInOverrides[childrenPropMapping] !== undefined) {
```

Comment:
Even if propsInOverrides[childrenPropMapping] === undefined we probably want to map for consistency sake.

Reply:
This would prevent the option of using `children` in overrides on components that have children prop mapping because the `undefined` children prop mapping would overwrite `children`.

```js
"overrides": {
  "Button": {
    "children": "componentWithVariantsWithChildrenProp"
  }
}
```
Converts to:
```js
"overrides": {
  "Button": {
    "children": undefined
  }
}
```

Reply:
hmmm, childrenPropMapping here is referring to the predefined prop that we want to map to children prop(Ex label). For example, we need to remap label prop to children prop for button component
```
 "overrides": {
        "Button": {
          "label": "ComponentWithVariantWithMappedChildrenProp",
          "fontSize": "15px"
        }
      }
```
convert to 

```
 "overrides": {
        "Button": {
          "children": "ComponentWithVariantWithMappedChildrenProp",
          "fontSize": "15px"
        }
      }
```

if children prop is not defined(propsInOverrides.children) and label prop is defined(propsInOverrides[childrenPropMapping], then we will remap label prop to children prop. So if propsInOverrides[childrenPropMapping] is undefined, then we don need to do anything as the label prop is not there.

Reply:
> Even if propsInOverrides[childrenPropMapping] === undefined we probably want to map for consistency sake.
hmmm, childrenPropMapping here is referring to the predefined prop that we want to map to children prop(Ex label). For example, we need to remap label prop to children prop for button component

 "overrides": {
        "Button": {
          "label": "ComponentWithVariantWithMappedChildrenProp",
          "fontSize": "15px"
        }
      }
convert to

 "overrides": {
        "Button": {
          "children": "ComponentWithVariantWithMappedChildrenProp",
          "fontSize": "15px"
        }
      }
if children prop is not defined(propsInOverrides.children) and label prop is defined(propsInOverrides[childrenPropMapping], then we will remap label prop to children prop. So if propsInOverrides[childrenPropMapping] is undefined, then we don need to do anything as the label prop is not there.